app-admin/lsat/files/lsat-0.9.8.2-segfault-fix.patch

```diff
@@ -0,0 +1,36 @@
+--- lsat-0.9.7.1/lsatmain.c.orig	2008-12-29 11:09:04.000000000 -0500
++++ lsat-0.9.7.1/lsatmain.c	2008-12-29 13:31:00.000000000 -0500
+@@ -308,9 +308,9 @@ int main(int argc, char *argv[])
+     */ 
+     char release[50]; /* array for release level		*/
+     char kernel[50];  /* what kernel user is running	*/
+-    static char *man_distro; /* if the user specifies a distribution */
++    static char man_distro[10]; /* if the user specifies a distribution */
+     const char * header   =NULL;   /* to print out the header */
+-    static char *out_file = "lsat.out";  /* output filename var  */
++    static char out_file[255] = "lsat.out";  /* output filename var  */
+     char xlist[100]; /* modules to exclude */
+     int xarray[33] = {0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0}; 
+     int  somethinginxlist = 0;
+@@ -350,10 +350,10 @@ int main(int argc, char *argv[])
+ 			  diff = 1; 
+ 			  break;
+ 		case 'm': 
+-			  strcpy(man_distro, argv[i]+3);
++			  strncpy(man_distro, argv[i]+3,sizeof(man_distro));
```

Comment:
I dare say silently truncating user arguments suck. If we don't care much about portability, how about `strdup()` instead?

Reply:
Perhaps you could suggest it upstream? However, it looks like the method employed by this patch is what the author used upstream, (See: https://github.com/triode3/lsat/blob/master/lsatmain.c#L357).

Reply:
I seriously do wonder how anyone can use 'security' in program name, and at the same time write horribly broken insecure code…